Storage/History.swift

```diff
@@ -40,6 +37,14 @@ public protocol BrowserHistory {
     func getPinnedTopSites() -> Deferred<Maybe<Cursor<Site>>>
 }
 
+/**
+ * An interface for fast repeated frecency queries.
+ */
+public protocol FrecentHistory {
+    func getSites(whereURLContains filter: String?, historyLimit limit: Int, bookmarksLimit: Int) -> Deferred<Maybe<Cursor<Site>>>
+    func updateTopSitesCacheQuery() -> (String, Args?)
```

Comment:
File a follow-up to replace this "give me your implementation details" method with "do this thing".